app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/transferdetails/TransferDetailsViewModel.kt

```diff
@@ -156,25 +169,60 @@ class TransferDetailsViewModel @Inject constructor(
             _isDeeplinkNeedingPassword.emit(false)
         }.cancellable().onFailure { exception ->
             when (exception) {
-                is ExpiredDateFetchTransferException -> Unit
-                is DownloadQuotaExceededException -> Unit
-                is NotFoundFetchTransferException -> longToast(R.string.deeplinkTransferNotFound)
-                is PasswordNeededFetchTransferException, is WrongPasswordFetchTransferException -> throw exception
+                is NotFoundFetchTransferException,
+                is ExpiredDateFetchTransferException,
+                is PasswordNeededFetchTransferException,
+                is WrongPasswordFetchTransferException,
+                is DownloadQuotaExceededException -> throw exception
                 else -> SentryLog.e(TAG, "An error has occurred when deeplink a transfer", exception)
             }
         }
     }
 
-    sealed class TransferDetailsUiState {
+    fun deleteTransfer(transferUuid: String) {
+        deleteTransfer(transferUuid, viewModelScope)
+    }
+
+    private fun TransferUi?.toUiState(): TransferDetailsUiState = when (this?.transferStatus) {
+        TransferStatus.READY, TransferStatus.UNKNOWN -> Success(this)
```

Comment:
I'm not sure I understand exactly what UNKOWN is supposed to represent. How do iOS handle the case of `TransferStatus.UNKNOWN`? Do they also treat it as a successful transfer? And why is this the choice we made?